backend/uclapi/uclapi/urls.py

```diff
@@ -26,6 +27,8 @@
     url(r'^dashboard/', include('dashboard.urls')),
     url(r'^docs', documentation),
     url(r'^about', about),
+    url(r'^settings', settings),
+    url(r'logout$', logout),
     url(r'^marketplace', marketplace),
```

Comment:
Ok, looking at the diff I've noticed I've noticed an issue.

First of all,  all these regexes I've commented on are wrong. Note that in Python regexes have the following format defined [here](https://docs.python.org/3/library/re.html). In particular you'll notice that `^` matches the start of the input and `$` matches the end of the input. Currently you'll be taken to the about page if you navigate to `about`, correctly, but also to `about123`, incorrectly. The inverse happens with `logout`, I will be taken the logout page if I navigate to `i-dont-want-to-logout`, ironically. The solution? All these regexes highlighted above require a `^` at the beginning and a `$` at the end. 

HOWEVER, only correct for the lines changed in this current diff. For the other lines that aren't changed please fix them in a separate PR. I would like to note that I believe `url()` is deprecated since Django 2.0 (which thanks to Zak, we've finally switched to), and `path()` is recommended instead. You can find a way to switch to `path()` [here](https://consideratecode.com/2018/05/02/django-2-0-url-to-path-cheatsheet/). I'll accept a diff that either uses `url()` correctly or switches to `path()` (also correctly), but `path() would be prefferred.

Reply:
Okay have switchedd to using Path for urls in this pull request and I will update the other ones in a future pull request too. Thanks for the heads up!